src/Avalonia.Controls/Primitives/Track.cs

```diff
@@ -126,19 +126,19 @@ protected override Size ArrangeOverride(Size finalSize)
             var viewportSize = ViewportSize;
             var extent = range + viewportSize;
 
-            if (Orientation == Orientation.Horizontal)
+            if (extent >= 0 && viewportSize >= 0)
             {
-                var thumbWidth = double.IsNaN(viewportSize) ? thumb?.DesiredSize.Width ?? 0 : finalSize.Width * viewportSize / extent;
-                var remaining = finalSize.Width - thumbWidth;
-                var firstWidth = range <= 0 ? 0 : remaining * offset / range;
-
-                if (decreaseButton != null)
+                if (Orientation == Orientation.Horizontal)
                 {
-                    decreaseButton.Arrange(new Rect(0, 0, firstWidth, finalSize.Height));
-                }
+                    var thumbWidth = double.IsNaN(viewportSize) ? thumb?.DesiredSize.Width ?? 0 : finalSize.Width * viewportSize / extent;
```

Comment:
is "double.IsNaN(viewportSize)" still needed? same line 154

Reply:
yes definitely needed, when its run through the unit tests, viewport size can be NaN.

latest code simplifies this line though.